doc/SEMANTICS.md

```diff
@@ -45,6 +45,9 @@ Not all S3 object keys correspond to valid file names, and these objects will no
 
 then mounting your bucket with Mountpoint will show only the `blue` directory, containing the file `image.jpg`. The `blue` object will not be accessible. See the [detailed semantics](#mapping-s3-object-keys-to-files-and-directories) below for more information about invalid object keys.
 
+> [!IMPORTANT]
+> Please note, that there is a known [issue](https://github.com/awslabs/mountpoint-s3/issues/725) with `readdir` operation (used to list files in a directory), which results in duplicate files in the response. Duplicate files may occur for any type of a bucket if it contains keys which should be hidden according to the semantics described above. E.g. for bucket with keys `blue` and `blue/image.jpg` two files with the name `blue` may be shown, and none of those may be a directory file.
```

Comment:
>and none of those may be a directory file.

Are we sure this is the case? I understand where we may see a directory and a file returned, but not two entries both for directories.